aptos-move/framework/aptos-stdlib/sources/cryptography/curves.move

```diff
@@ -0,0 +1,217 @@
+module aptos_std::curves {
+    use aptos_std::type_info::type_of;
+
+    // Structs and consts.
+
+    // Fake structs representing group type.
+    struct BLS12_381_G1 {}
+    struct BLS12_381_G2 {}
+    struct BLS12_381_Gt {}
+
+    //TODO: handle as u8 temporarily. Upgrade to u64.
+    struct Scalar<phantom Group> has drop {
+        handle: u8
+    }
+
+    struct Point<phantom Group> has drop {
+        handle: u8
+    }
+
+    /// Get internal handle for a Scalar. Currently needed by groth16 module.
+    /// TODO: can this be avoided?
+    public fun get_scalar_handle<Group>(s: &Scalar<Group>): u8 {
+        s.handle
+    }
+
+    /// Get internal handle for a point. Currently needed by groth16 module.
+    /// TODO: can this be avoided?
+    public fun get_point_handle<Group>(p: &Point<Group>): u8 {
+        p.handle
+    }
+
+    /// Perform a bilinear mapping.
+    /// TODO: is it possible to have 2+ mappings between same (G1,G2,Gt)? If so we need a parameter for `mapping_id`?
```

Comment:
Do you mean if it's possible to have more than 1 bilinear map across the same G_1, G_2 groups? Maybe with different G_{T_i}'s for each mapping i?

That's a neat question. I am not aware of any such pairs of curves, so for now, let's assume the answer is "no."